Updating documentation. (#58)
kev-m/pyCoilGen: docs/source/configuration.md

* The [coil mesh](#coil-meshes) defines the surface upon which the coil winding path must be computed. This geometry is required.  

* The [target field](#target-field) is a vector field of co-ordinates and magnitudes. This geometry is required. It may either be specified by loading an existing
target field, which provides both the co-ordinates and the magnitudes at each co-ordinate, or by separately specifying
the co-ordinates and a gradient field equation.

* The [shield mesh](#shield-mesh) is an optional geomerty which defines an additional surface where the magnetic field must be suppressed.

These mesh geometries can either be loaded from a [pre-optimised file](#pre-calculated-mesh-and-stream-function) or specified individually using [mesh creation builders](#mesh-creation-builders).

### Coil Meshes

The purpose of `pyCoilGen` is to generate a coil wire path on a coil mesh surface or surfaces to produce a desired target field.

The coil mesh surface can be specified using a coil mesh builder and a corresponding builder configuration parameter.

- `coil_mesh` (Type: `str`, Default: `'none'`)

  Define the coil surface for the wire path.

>   Specify the [mesh builder](#mesh-creation-instructions) to create the coil mesh surface(s). The corresponding creation builder parameter must also be provided.

Reviewer: ```suggestion
  Specify the [mesh builder](#mesh-creation-builders) to create the coil mesh surface(s). The corresponding creation builder parameter must also be provided.
```

- `coil_mesh_file` (Type: `str`, Default: `'none'`) (*deprecated*)

>   Either specify the filename of an `.stl` file to be loaded from `geometry_source_path`, or use one of the built-in mesh specifications. When using a built-in mesh specification, the mesh parameters must also be specified.

Reviewer: What is a "built-in mesh specification"?

  Circular diameter factor for projecting the 3D coil mesh to 2D.

### Target Field

The purpose of `pyCoilGen` is to generate coils that produce a desired target field. This target field could be a
gradient field or a generic target field.

The target vector field can either be specified using a mesh or a sphere to define the co-ordinates and the 
`field_shape_function` to define the magnitudes, or loaded from a NumPy pickle file. The NumPy pickle file takes
precedence, if specified.

A gradient field is specified by using a field shape function.

#### Specifying the Target Field Co-ordinates Using a Mesh

The mesh defines the boundary of the target field and these parameters fine-tune the target field point selection.

- `target_mesh` (Type: `str`, Default: `'none'`)

>   Specify the mesh [creation instruction](#mesh-creation-instructions) to create the target field co-ordinates. The

Reviewer: ```suggestion
  Specify the [mesh creation builder](#mesh-creation-builders) to create the target field co-ordinates. The
```

>   corresponding creation instruction parameter must also be provided.

Reviewer: ```suggestion
  corresponding builder creation parameter must also be provided.
```

  }
  solution = pyCoilGen(log, parameters)
```

### Shield Mesh

The optional shield mesh specifies where the magnetic field must be zero, for example for suppressed outer regions in
active shields.

- `shield_mesh` (Type: `str`, Default: `'none'`)

>   Specify the mesh [creation instruction](#mesh-creation-instructions) to create the shield mesh co-ordinates. The 

Reviewer: ```suggestion
  Specify the mesh [creation instruction](#mesh-creation-builders) to create the shield mesh co-ordinates. The 
```

>   corresponding creation instruction parameter must also be provided.

Reviewer: ```suggestion
  corresponding builder parameter must also be provided.
```

        center_position_y (`float`): Y-coordinate of the center position.
        center_position_z (`float`): Z-coordinate of the center position.
        plane_distance (`float`): Distance between the two planes.

- `create circular mesh`
  Create a circular mesh defined by `circular_mesh_parameter_list` (Type: `list of numeric`, Default: `[0.25, 20, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]`)

        radius (`float`): Radius of the mesh.
        num_radial_divisions (`int`): Number of divisions in the radial direction.
        rotation_vector_x (`float`): X component of the rotation vector.
        rotation_vector_y (`float`): Y component of the rotation vector.
        rotation_vector_z (`float`): Z component of the rotation vector.
        rotation_angle (`float`): Rotation angle in radians.
        center_position_x (`float`): X component of the center position.
        center_position_y (`float`): Y component of the center position.
        center_position_z (`float`): Z component of the center position.

- `create stl mesh`
  Create the mesh from the file specified with `stl_mesh_filename` (Type: `str`, Default: `none`)

>   Numerous file types are supported: STL, GLB, PLY, 3MF, XAML, etc. The mesh is loaded from the `geometry_source_path`

Reviewer: It will be better to list all supported types.
